File: migration-tools/create-tables.php
<?php

/**
 * Script para criar tabelas no Supabase
 * Feature: deploy-and-supabase-migration
 */

require_once __DIR__ . '/DatabaseConnection.php';

// Carregar variáveis de ambiente
function loadEnv($file) {
    if (!file_exists($file)) {
        die("Erro: Arquivo .env não encontrado!\n");
    }
    
    $lines = file($file, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    foreach ($lines as $line) {
        if (strpos(trim($line), '#') === 0) continue;
        
        list($name, $value) = explode('=', $line, 2);
        $_ENV[trim($name)] = trim($value);
    }
}

loadEnv(__DIR__ . '/../.env');

// Configuração do Supabase
$config = [
    'host' => $_ENV['SUPABASE_HOST'],
    'port' => $_ENV['SUPABASE_PORT'],
    'database' => $_ENV['SUPABASE_DATABASE'],
    'user' => $_ENV['SUPABASE_USER'],
    'password' => $_ENV['SUPABASE_PASSWORD']
];

echo "=== Criação de Tabelas no Supabase ===\n\n";

// Testar conexão
echo "1. Testando conexão com Supabase...\n";
$db = new DatabaseConnection($config);

if (!$db->testConnection()) {
    die("❌ Erro: Não foi possível conectar ao Supabase!\n");
}
echo "✅ Conexão estabelecida com sucesso!\n\n";

// Ler schema PostgreSQL
echo "2. Lendo schema PostgreSQL...\n";
$schemaFile = __DIR__ . '/sql_subway_pg.sql';
if (!file_exists($schemaFile)) {
    die("❌ Erro: Arquivo sql_subway_pg.sql não encontrado!\n");
}

$schema = file_get_contents($schemaFile);
echo "✅ Schema carregado!\n\n";

// Executar schema
echo "3. Criando tabelas no Supabase...\n";
$results = $db->executeMultiple($schema);

$successCount = 0;
$errorCount = 0;

foreach ($results as $result) {
    if ($result['success']) {
        $successCount++;
        echo "✅ " . $result['statement'] . "\n";
    } else {
        $errorCount++;
        echo "❌ " . $result['statement'] . "\n";
        echo "   Erro: " . $result['error'] . "\n";
    }
}

echo "\n4. Validando tabelas criadas...\n";
$tables = $db->getTables();
echo "✅ Tabelas encontradas: " . count($tables) . "\n";

foreach ($tables as $table) {
    $count = $db->countRecords($table);
    echo "   - $table: $count registros\n";
}

echo "\n=== Resumo ===\n";
echo "✅ Statements executados com sucesso: $successCount\n";
if ($errorCount > 0) {
    echo "❌ Statements com erro: $errorCount\n";
}
echo "✅ Total de tabelas: " . count($tables) . "\n";

if ($errorCount === 0) {
    echo "\n🎉 Todas as tabelas foram criadas com sucesso!\n";
} else {
    echo "\n⚠️  Algumas operações falharam. Verifique os erros acima.\n";
}

File: migration-tools/create-tables.py
#!/usr/bin/env python3
"""
Script para criar tabelas no Supabase (alternativa Python)
Feature: deploy-and-supabase-migration
"""

import os
import psycopg2
from psycopg2 import sql

def load_env(file_path):
    """Carrega variáveis de ambiente do arquivo .env"""
    env_vars = {}
    with open(file_path, 'r') as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#'):
                key, value = line.split('=', 1)
                env_vars[key.strip()] = value.strip()
    return env_vars

def main():
    print("=== Criação de Tabelas no Supabase ===\n")
    
    # Carregar credenciais
    env_file = os.path.join(os.path.dirname(__file__), '..', '.env')
    env = load_env(env_file)
    
    # Configuração de conexão
    conn_params = {
        'host': env['SUPABASE_HOST'],
        'port': env['SUPABASE_PORT'],
        'database': env['SUPABASE_DATABASE'],
        'user': env['SUPABASE_USER'],
        'password': env['SUPABASE_PASSWORD'],
        'sslmode': 'require'
    }
    
    # Testar conexão
    print("1. Testando conexão com Supabase...")
    try:
        conn = psycopg2.connect(**conn_params)
        print("✅ Conexão estabelecida com sucesso!\n")
    except Exception as e:
        print(f"❌ Erro ao conectar: {e}")
        return
    
    # Ler schema
    print("2. Lendo schema PostgreSQL...")
    schema_file = os.path.join(os.path.dirname(__file__), 'sql_subway_pg.sql')
    with open(schema_file, 'r', encoding='utf-8') as f:
        schema = f.read()
    print("✅ Schema carregado!\n")
    
    # Executar schema
    print("3. Criando tabelas no Supabase...")
    cursor = conn.cursor()
    
    # Dividir em statements
    statements = [s.strip() for s in schema.split(';') if s.strip()]
    
    success_count = 0
    error_count = 0
    
    for statement in statements:
        if not statement:
            continue
            
        try:
            cursor.execute(statement)
            conn.commit()
            success_count += 1
            preview = statement[:100].replace('\n', ' ')
            print(f"✅ {preview}...")
        except Exception as e:
            error_count += 1
            preview = statement[:100].replace('\n', ' ')
            print(f"❌ {preview}...")
            print(f"   Erro: {e}")
    
    # Validar tabelas
    print("\n4. Validando tabelas criadas...")
    cursor.execute("""
        SELECT table_name 
        FROM information_schema.tables 
        WHERE table_schema = 'public' 
        AND table_type = 'BASE TABLE'
        ORDER BY table_name
    """)
    
    tables = cursor.fetchall()
    print(f"✅ Tabelas encontradas: {len(tables)}\n")
    
    for (table,) in tables:
        cursor.execute(f'SELECT COUNT(*) FROM "{table}"')
        count = cursor.fetchone()[0]
        print(f"   - {table}: {count} registros")
    
    # Resumo
    print("\n=== Resumo ===")
    print(f"✅ Statements executados com sucesso: {success_count}")
    if error_count > 0:
        print(f"❌ Statements com erro: {error_count}")
    print(f"✅ Total de tabelas: {len(tables)}")
    
    if error_count == 0:
        print("\n🎉 Todas as tabelas foram criadas com sucesso!")
    else:
        print("\n⚠️  Algumas operações falharam. Verifique os erros acima.")
    
    cursor.close()
    conn.close()

if __name__ == '__main__':
    main()
